controllers/util/common/common.go

```diff
@@ -413,3 +427,44 @@ func (ch *ControllerHelper) SetS3ProviderEP() {
 		ch.CosEP = fmt.Sprintf(constants.WasabiEP, ch.S3ProviderRegion)
 	}
 }
+
+// Update finalizer to Controller Deployment under NS "ibm-object-csi-operator"
+// op = 1  Add finalizer    op = 2  Remove finalizer
+func (ch *ControllerHelper) updateControllerFinalizer(ctx context.Context, op constants.FinalizerOps, finalizerName string) error {
+	ch.Log.Info("updateControllerFinalizer(): Entry")
+	defer ch.Log.Info("updateControllerFinalizer(): Exit")
+
+	ctrlDep := &appsv1.Deployment{}
+	err := ch.Get(ctx, client.ObjectKey{Namespace: constants.CSIOperatorNamespace, Name: constants.DeploymentName}, ctrlDep)
+	if err != nil {
+		ch.Log.Error(err, "updateControllerFinalizer(): Controller Deployment not found")
+		return err
+	}
+
+	if op == constants.AddFinalizer { // Add finalizer
+		ch.Log.Info("updateControllerFinalizer(): Add finalizer to Controller deployment")
+		chk := controllerutil.ContainsFinalizer(ctrlDep, finalizerName)
+		if !chk {
+			controllerutil.AddFinalizer(ctrlDep, finalizerName)
+			err = ch.Update(ctx, ctrlDep)
+		} else {
+			ch.Log.Info("updateControllerFinalizer(): Finalizer already applied to Controller deployment")
+		}
+	}
+
+	if op == constants.RemoveFinalizer { // Remove finalizer
+		ch.Log.Info("updateControllerFinalizer(): Remove finalizer from Controller deployment")
+		chk := controllerutil.ContainsFinalizer(ctrlDep, finalizerName)
+		if chk {
+			controllerutil.RemoveFinalizer(ctrlDep, finalizerName)
+			err = ch.Update(ctx, ctrlDep)
+		} else {
+			ch.Log.Info("updateControllerFinalizer(): Finalizer already removed from Controller deployment")
+		}
+	}
+	if err != nil {
```

Comment:
Lets remove this err check from here and move it inside both conditional checks.
Then it would be easier to log whether addition/deletion operator completed successfully.